packages/docs/celo-codebase/protocol/identity/phone-number-privacy.md

```diff
@@ -0,0 +1,38 @@
+# Phone Number Privacy
+
+An important part of the Celo protocol is the ability to link a phone number to a Celo native address. This improves the user experience for sending/receiving payments to a phone's contacts without having to deal with blockchain native addresses. However, it's also important that the phone number mappings remain private, preventing the mass harvesting of mapped phone numbers. This guide provides an overview of the Celo phone number privacy system.
+
+## Understanding the problem
+When you send payment to someone in your phone's address book, the mobile client needs to look up the phone number in the on-chain registry to find the mapping to a Celo native address, which can be used to receive the actual payment. If this mapping were in plain text on the Celo blockchain then anyone would be able to trivially see account balances for all numbers. If instead, the hash of each phone number were used as the on chain identifier, it would still be possible to brute force the mapping via a [rainbow table attack](https://en.wikipedia.org/wiki/Rainbow_table). 
+
+## The "pretty good" solution
+The basis of the solution is to use a hashed on-chain identifier that is derived from both the original phone number and a salt that is provided by the Pretty Good Phone Number Privacy (PGPNP) service. When a user initiates a payment to a phone number, the mobile wallet first queries PGPNP for the number's salt and then uses this salt to compute the on-chain identifier. When the PGPNP receives a request, it authenticates the request and ensures the account has not exceeded its quota. Since blockchain accounts and phone numbers are not naturally Sybil-resistant, the PGPNP uses a set of factors to determine the appropriate quota. 
```

Comment:
lol -- do we want to refer to it as a "pretty good" phone number privacy service externally?

Reply:
Haha I'll bring it up in the slack and we can discuss. I'm in favor of the name including "pretty good" because it shows that there are some limitations to the privacy (doesn't prevent targeted attacks).